### PR TITLE
Fix tests on macOS and update coverage config

### PR DESCRIPTION
**Pull request by phoerious**
1. Fixes hanging tests on macOS
2. Simplifies coverage generation and uses llvm-cov instead of gcov if compiler is clang (which is both faster and prettier).

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Review by droidmonkey — APPROVED** [review]

**Comment by phoerious** [comment]
@droidmonkey Ready for re-review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -394,16 +385,44 @@ endif(WITH_TESTS)
...
     append_coverage_compiler_flags()
-    setup_target_for_coverage_gcovr_html(
-            NAME coverage
-            EXECUTABLE $(MAKE) && $(MAKE) test
-    )
+
+    if(WITH_COVERAGE AND CMAKE_COMPILER_IS_CLANGXX)
+        if(APPLE AND WITH_APP_BUNDLE)
+            set(MAIN_BINARIES
+                    "${CMAKE_BINARY_DIR}/src/${BIN_INSTALL_DIR}/${PROGNAME}"
```
Comment (droidmonkey) [inline CMakeLists.txt]: We really should be using generator expressions for these https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:TARGET_FILE

**Review by droidmonkey — APPROVED** [review]
Approving with comment

File: CMakeLists.txt (hunk 1 above)
Comment (phoerious) [reply CMakeLists.txt]: Good idea.

Comment (phoerious) [reply CMakeLists.txt]: Done. Much cleaner already.

**Review by droidmonkey — APPROVED** [review]

**Pull request merged**